airbyte-cdk/bulk/toolkits/extract-cdc/src/main/kotlin/io/airbyte/cdk/read/cdc/DebeziumOperations.kt

```diff
@@ -23,6 +23,15 @@ interface CdcPartitionsCreatorDebeziumOperations<T : Comparable<T>> {
 
     /** Builds a [DebeziumInput] using an incumbent [OpaqueStateValue]. */
     fun deserialize(opaqueStateValue: OpaqueStateValue, streams: List<Stream>): DebeziumInput
+
+    /* Validate the input state is still valid on DB. */
+    fun validate(debeziumInput: DebeziumInput): CdcStateValidateResult
```

Comment:
Is this method really necessary? Couldn't validation could be the responsibility of `deserialize`?

Perhaps I'm missing but I don't think you'd even need `CdcStateValidateResult`:
- for `VALID` let `deserialize` return the deserialized `DebeziumInput`,
- for `INVALID_ABORT` let `deserialize` throw an exception,
- for `INVALID_RESET` let `deserialize` grumble about resetting in the logs and delegate to `synthetic`.

Would this work? You still need to call `markCdcAsComplete` on throwing but you could wrap the call to `deserialize()` in a try-catch block and rethrow after calling `markCdcAsComplete`.

Reply:
thanks yeah that'll be great! My intention was to just create separate function for clarity - but yeah I think this can also be viewed as an internal check within `deserialize` and does not have to be exposed.